ecl/resource_ecl_network_load_balancer_v2.go

```diff
@@ -0,0 +1,1384 @@
+			id, err)
+	}
+
+	// also wait for Load Balancer Core
+	stateConf = &resource.StateChangeConf{
+		Pending:      []string{"PENDING_UPDATE"},
+		Target:       []string{"ACTIVE"},
+		Refresh:      waitForLoadBalancerComplete(networkClient, d.Id()),
+		Timeout:      d.Timeout(schema.TimeoutUpdate),
+		Delay:        5 * time.Second,
+		PollInterval: loadBalancerUpdatePollInterval,
+		MinTimeout:   3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"error waiting for Load Balancer (%s) to become ACTIVE(after interface update): %s",
+			d.Id(), err)
+	}
+
+	return nil
+}
+
+func updateLoadBalancerSyslogServer(networkClient *eclcloud.ServiceClient, d *schema.ResourceData, id string, updateOpts load_balancer_syslog_servers.UpdateOpts) error {
+	log.Printf("[DEBUG] Updating Load Balancer Syslog Server %s with options: %+v", id, updateOpts)
+	_, err := load_balancer_syslog_servers.Update(networkClient, id, updateOpts).Extract()
+	if err != nil {
+		return fmt.Errorf(
+			"error updating for Load Balancer Syslog Server (%s) with option %#v: %s",
+			id, updateOpts, err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING_UPDATE"},
+		Target:       []string{"ACTIVE"},
+		Refresh:      waitForLoadBalancerSyslogServerComplete(networkClient, id),
+		Timeout:      d.Timeout(schema.TimeoutUpdate),
+		Delay:        5 * time.Second,
+		PollInterval: loadBalancerUpdatePollInterval,
+		MinTimeout:   3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"error waiting for Load Balancer Syslog Server (%s) to become ACTIVE(after Syslog Server update): %s",
+			id, err)
+	}
+
+	return nil
+}
+
+func getLoadBalancerUpdateOpts(d *schema.ResourceData, gatewayInitialized bool, planUpdated bool) *load_balancers.UpdateOpts {
+	var returnUpdateOpts bool
+
+	var updateOpts load_balancers.UpdateOpts
+
+	s := d.Get("default_gateway").(string)
+	if d.HasChange("default_gateway") || (gatewayInitialized && s != "") {
+		returnUpdateOpts = true
+		var defaultGateway interface{}
+		if s == "" {
+			defaultGateway = nil
+		} else {
+			defaultGateway = s
+		}
+		updateOpts.DefaultGateway = &defaultGateway
+	}
+
+	if d.HasChange("description") {
+		returnUpdateOpts = true
+		description := d.Get("description").(string)
+		updateOpts.Description = &description
+	}
+
+	if d.HasChange("load_balancer_plan_id") && !planUpdated {
+		returnUpdateOpts = true
+		loadBalancerPlanID := d.Get("load_balancer_plan_id").(string)
+		updateOpts.LoadBalancerPlanID = loadBalancerPlanID
+	}
+
+	if d.HasChange("name") {
+		returnUpdateOpts = true
+		name := d.Get("name").(string)
+		updateOpts.Name = &name
+	}
+
+	if returnUpdateOpts {
+		return &updateOpts
+	} else {
+		return nil
+	}
+}
+
+func getLoadBalancerInterfaceInitialUpdateOpts(interfaceConfig map[string]interface{}) *load_balancer_interfaces.UpdateOpts {
```

Comment:
terraformのconfigをgoの変数に保存する関数は一般的にexpandという動詞を使います。
```suggestion
func expandLoadBalancerInterfaceInitialUpdateOpts(interfaceConfig map[string]interface{}) *load_balancer_interfaces.UpdateOpts {
```

Reply:
expand～という関数名に変更しました。